Objects/longobject.c

```diff
@@ -5879,7 +5879,7 @@ int___sizeof___impl(PyObject *self)
 {
     Py_ssize_t res;
 
-    res = offsetof(PyLongObject, ob_digit) + Py_ABS(Py_SIZE(self))*sizeof(digit);
+    res = offsetof(PyLongObject, ob_digit) + Py_MAX(Py_ABS(Py_SIZE(self)), 1)*sizeof(digit);
```

Comment:
Minor suggestion: break this up to avoid the long line (we try but don't always succeed to keep line length to 79 characters), and add a comment explaining what's going on:

E.g.:
```suggestion
    res = offsetof(PyLongObject, ob_digit)
        /* using Py_MAX(..., 1) because we always allocate space for at least
           one digit, even though the integer zero has a Py_SIZE of 0 */
        + Py_MAX(Py_ABS(Py_SIZE(self)), 1)*sizeof(digit);
```